core/src/main/java/com/cryptomorin/xseries/XItemBuilder.java

```diff
@@ -0,0 +1,437 @@
+    public XItemBuilder withUnbreakable(boolean unbreakable) { return property(new Unbreakable(unbreakable)); }
+    public XItemBuilder withCustomModelData(int customModelData) { return property(new CustomModelData(customModelData)); }
+    public XItemBuilder withLore(List<String> lore) { return property(new Lore(lore)); }
+    public XItemBuilder withBookAuthor(String bookAuthor) { return property(new BookAuthor(bookAuthor)); }
+
+
+    public static final class Material extends LambdaProperty<XMaterial> {
+        public Material(XMaterial material) { super(material, null, (a, b) -> {}, XMaterial::matchXMaterial); }
+        public Material() { this(null); }
+
+        public XMaterial getMaterial() { return super.value; }
+    }
+
+    public static final class Amount extends LambdaProperty<Integer> {
+        private static final int DEFAULT_VALUE = 1;
+
+        public Amount(Integer amount) { super(amount, DEFAULT_VALUE, ItemStack::setAmount, ItemStack::getAmount); }
+        public Amount() { this(DEFAULT_VALUE); }
+    }
+
+    public static final class DisplayName extends LambdaMetaProperty<ItemMeta, String> {
+        public DisplayName(String displayName) {
+            super(displayName, null, () -> ItemMeta.class, ItemMeta::setDisplayName,
+                    conditional(ItemMeta::hasDisplayName, ItemMeta::getDisplayName));
+        }
+        public DisplayName() { this(null); }
+    }
+
+    @SuppressWarnings("deprecation")
+    public static class Durability implements Property {
+        private static final boolean SUPPORTS_META = checkMetaAvailable("Damageable");
+        private int durability;
+
+        public Durability() {}
+        public Durability(int durability) { this.durability = durability; }
+
+        @Override public void to(ItemStack item, ItemMeta meta) {
+            if (SUPPORTS_META) {
+                if (meta instanceof Damageable) {
+                    ((Damageable) meta).setDamage(durability);
+                }
+            } else {
+                item.setDurability((short) durability);
+            }
+        }
+
+        @Override public void from(ItemStack item, ItemMeta meta) {
+            if (SUPPORTS_META) {
+                if (meta instanceof Damageable) {
+                    durability = ((Damageable) meta).getDamage();
+                }
+            } else {
+                durability = item.getDurability();
+            }
+        }
+
+        @Override public boolean affectsMeta() { return SUPPORTS_META; }
+        @Override public boolean isDefault() { return durability == 0; }
+    }
+
+    public static final class Unbreakable extends LambdaMetaProperty<ItemMeta, Boolean> {
+        public Unbreakable(final Boolean unbreakable) {
+            super(unbreakable, false, () -> ItemMeta.class, ItemMeta::setUnbreakable, ItemMeta::isUnbreakable);
+        }
+        public Unbreakable() { this(false); }
+
+        @Override public boolean isSupported() { return supports(11); }
```

Comment:
Should we use `XMaterial.supports` here? Or check with reflections for the method(s) used by this property.

Reply:
You should check with reflection.